crates/parry2d/examples/raycasts_animated.rs

```diff
@@ -0,0 +1,124 @@
+use macroquad::prelude::*;
+use nalgebra::{Isometry2, Point2, UnitComplex, Vector2};
+use parry2d::query::{Ray, RayCast};
+use parry2d::shape::{Cuboid, TriMesh};
+
+const RENDER_SCALE: f32 = 30.0;
+
+#[macroquad::main("parry2d::utils::polygons_intersection_points")]
+async fn main() {
+    let animation_scale = 2.0;
+    let animation_rotation = UnitComplex::new(0.008);
+
+    let spikes_render_pos = Point2::new(300.0, 300.0);
+
+    for i in 1.. {
+        clear_background(BLACK);
+
+        /*
+         *
+         * Compute the rotated/scaled cuboid.
+         *
+         */
+        let mut animated_cube = Cuboid::new(Vector2::new(2.0, 2.0)).to_polyline();
+        animated_cube.iter_mut().for_each(|pt| {
+            *pt = animation_rotation.powf(i as f32)
+                * *pt
+                * ((i as f32 / 100.0).sin().abs() * animation_scale);
+        });
+        let trimesh = TriMesh::from_polygon(animated_cube).unwrap();
```

Comment:
This looks a bit convoluted. Why not cast the ray on the `Cuboid` directly? (without conversion)
Regarding the rotation, it would be in the `Isometry2` given as the first argument of `.cast_ray` instead of being applied to the geometry directly.
And the scale can be applied to the `Cuboid` half-extents directly.
I suppose this might have been copied from the trimesh intersection algorithm which did need that trimesh conversion due to the nature of the code being tested. It’s not needed for this test.